langchain/schema/language_model.py

```diff
@@ -48,6 +69,161 @@ class BaseLanguageModel(Serializable, ABC):
+        else:
+            raise ValueError(
+                f"Invalid input type {type(input)}. "
+                "Must be a PromptValue, str, or list of BaseMessages."
+            )
+
+    def invoke(
+        self,
+        input: LanguageModelInput,
+        stop: Optional[List[str]] = None,
+        **kwargs: Unpack[RunnableConfig],
+    ) -> str:
+        return (
+            self.generate_prompt([self._convert_input(input)], stop=stop, **kwargs)
+            .generations[0][0]
+            .text
+        )
+
+    async def ainvoke(
+        self,
+        input: LanguageModelInput,
+        stop: Optional[List[str]] = None,
+        **kwargs: Unpack[RunnableConfig],
+    ) -> str:
+        llm_result = await self.agenerate_prompt(
+            [self._convert_input(input)], stop=stop, **kwargs
+        )
+        return llm_result.generations[0][0].text
+
+    def batch(
+        self,
+        inputs: List[LanguageModelInput],
+        config: Optional[RunnableConfig | List[RunnableConfig]] = None,
+        max_concurrency: Optional[int] = None,
+    ) -> List[str]:
+        if isinstance(config, list):
+            config = config[0]
+        if config is None:
+            config = {}
+
+        llm_result = self.generate_prompt(
+            [self._convert_input(input) for input in inputs], **config
+        )
+        return [g[0].text for g in llm_result.generations]
+
+    async def abatch(
+        self,
+        inputs: List[LanguageModelInput],
+        config: Optional[RunnableConfig | List[RunnableConfig]] = None,
+        max_concurrency: Optional[int] = None,
+    ) -> List[str]:
+        if isinstance(config, list):
+            config = config[0]
+        if config is None:
+            config = {}
+
+        llm_result = await self.agenerate_prompt(
+            [self._convert_input(input) for input in inputs], **config
+        )
+        return [g[0].text for g in llm_result.generations]
+
+    def stream(
+        self,
+        input: LanguageModelInput,
+        stop: Optional[List[str]] = None,
+        **kwargs: Unpack[RunnableConfig],
+    ) -> Iterator[str]:
+        if not hasattr(self, "streaming"):
+            # model doesn't support streaming, so use default implementation
+            yield self.invoke(input, stop=stop, **kwargs)
+        else:
+            from langchain.callbacks.streaming_iter import IteratorCallbackHandler
+
+            # enable streaming, if it's not already enabled
+            original_streaming = cast(bool, self.streaming)  # type: ignore
+            self.streaming = True
+
+            # add iter callback handler to kwargs
+            callbacks: Optional[Callbacks] = kwargs.pop("callbacks", None)
+            callback_handler = IteratorCallbackHandler()
+            if callbacks is None:
+                kwargs["callbacks"] = [callback_handler]
+            elif isinstance(callbacks, list):
+                callbacks.append(callback_handler)
```

Comment:
it's intentional that this is no longer part of kwargs?

Reply:
Oops good point